contentcuration/contentcuration/assets/channelList/utils.js

```diff
@@ -0,0 +1,39 @@
+import omit from 'lodash/omit';
+import cloneDeep from 'lodash/cloneDeep';
+import isEqual from 'lodash/isEqual';
+import store from './store';
+
+export function lastSavedStateFactory(idKey = 'id') {
+  // Return an object with methods to track the last saved state
+  const lastSavedStore = {};
+  return {
+    storeLastSavedState(object) {
+      // Store a record of previous attributes here for dirty checking.
+      lastSavedStore[object[idKey]] = cloneDeep(object);
+    },
+    hasUnsavedChanges(object) {
+      return !isEqual(object, lastSavedStore[object[idKey]]);
+    },
+    getUnsavedChanges(object) {
```

Comment:
The main reason for this is to track unsaved changes to allow two things:
* Conditional save behaviour, so that we only let the user make saves when changes have been made (and we can alert the user to unsaved changes).
* Diffed PATCHing - so that we can determine which top level keys of an object have changed, so that we only need to send the diff in the payload to the backend, reducing the size of database update operations because we use as few fields to update as possible.